driver/src/orderbook/auction_data_reader.rs

```diff
@@ -136,9 +95,7 @@ impl PaginatedAuctionDataReader {
         packed_auction_bytes
             .chunks(AUCTION_ELEMENT_WIDTH)
             .map(|chunk| {
-                let mut chunk_array = [0; AUCTION_ELEMENT_WIDTH];
-                chunk_array.copy_from_slice(chunk);
-                let mut result = StableXAuctionElement::from_bytes(&chunk_array);
+                let mut result = auction_element_from_chunk(chunk);
```

Comment:
naming nit: I would call this `..._from_slice` or `..._from_bytes` to be more consistent with Rust naming.